presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveViews.java

```diff
@@ -295,6 +297,40 @@ public void testSelectFromHiveViewWithoutDefaultCatalogAndSchema()
                 .containsExactly(row(1L));
     }
 
+    @Test
+    public void testTimestampHiveView() {
+        onHive().executeQuery("DROP TABLE IF EXISTS timestamp_hive_table");
+        onHive().executeQuery("CREATE TABLE timestamp_hive_table AS SELECT cast('1990-01-02 12:13:14.123456789' AS timestamp) ts");
+        onHive().executeQuery("DROP VIEW IF EXISTS timestamp_hive_view");
+        onHive().executeQuery("CREATE VIEW timestamp_hive_view AS SELECT * FROM timestamp_hive_table");
+
+        // timestamp_precision not set UNSET
+        assertThat(query("SELECT CAST(ts as VARCHAR) FROM timestamp_hive_view")).containsExactly(row("1990-01-02 12:13:14.123"));
+        assertThatThrownBy(
+                // TODO it is not possible to query Hive view with timestamps if hive.timestamp-precision=NANOSECONDS
+                () -> assertThat(query("SELECT CAST(ts as VARCHAR) FROM hive_timestamp_nanos.default.timestamp_hive_view")).containsExactly(row("1990-01-02 12:13:14.123456789"))
+        ).hasMessageContaining("timestamp(9) projected from query view at position 0 cannot be coerced to column [ts] of type timestamp(3) stored in view definition");
```

Comment:
https://github.com/prestosql/presto/issues/6295 is worded as an enhancement ("support this with that"), but the failure looks more like a bug. Maybe we do not need to create separate bug issue, but would be worth adding a comment in this issue, sth like "Currently this fails: «stacktrace»"

Reply:
https://github.com/prestosql/presto/issues/6295#issuecomment-749605985